### ISSUE TITLE
Get rid of false negative in `seq_linter()`

### ISSUE THREAD

**Issue by IndrajeetPatil** [r-lib/lintr#1474]
This should produce a lint:

``` r
library(lintr)

lint(
  text = "seq(dim(x)[1])",
  linter = seq_linter()
)
```

<sup>Created on 2022-07-26 by the [reprex package](https://reprex.tidyverse.org) (v2.0.1)</sup>

Because

``` r
seq(dim(data.frame())[1])
#> [1] 1 0
```

**Comment by IndrajeetPatil** [comment]
Test to use:

```r
expect_lint(
    "function(x) { seq(dim(x)[1]) }",
    rex("seq(dim(...))", anything, "Use seq_len(...)"),
    linter
)
```

**Comment by IndrajeetPatil** [comment]
I saw this test, which should actually fail:

https://github.com/r-lib/lintr/blob/a99b3c185435b6bf3cde162a3f6a891c7e30f635/tests/testthat/test-seq_linter.R#L101-L107

Since this is actually fine even in empty edge cases:

``` r
length(data.frame()):1
#> [1] 0 1

length(data.frame()):0
#> [1] 0
```

<sup>Created on 2022-07-26 by the [reprex package](https://reprex.tidyverse.org) (v2.0.1)</sup>

**Comment by MichaelChirico** [comment]
sorry, I don't follow why `length(data.frame()):1` is ok. the reasoning is the same as for `1:length(data.frame())`. The point is the is usually used to index columns, but `1` is not a valid column number for a 0-column data.frame.

So

```r
for (jj in length(myDF):1) {
  myDF[[jj]] = foo(myDF, jj + 1) # some recursive definition depending on previous columns
}
```

is likely a mistake when `ncol(myDF)` is 0

**Comment by IndrajeetPatil** [comment]
> sorry, I don't follow why length(data.frame()):1 is ok. 

I am not saying it is okay. I just saw this already existing test and realized that it is wrong, and so wanted to keep track of it in this issue.

**Comment by MichaelChirico** [comment]
Sorry, still not following why the test is wrong
